### PR TITLE
FAWS Examples

### PR DESCRIPTION
**Pull request by bradgignac**
This is a work-in-progress PR that adds FAWS-specific examples to the IdF user guide in preparation for the upcoming beta launch for FAWS. More information coming soon.

### CODE REVIEW

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/build-1fc7e2b12f/docs/rackspace-federation/). :bow:

File: doc/attribmapping-basics/faws-mapping.rst [hunk 1]
```diff
@@ -1,9 +1,149 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These permissions control access to features within Rackspace's Fanatical Support for AWS Control Panel. User permissions can be set to ``observer``, ``admin``, or omitted from the mapping policy. Users with ``observer`` permissions have read-only access to the control panel. Users with ``admin`` permissions have read and write access to the control panel. These permissions are assigned by granting them as roles in a mapping policy. The following mapping policy assigns the ``admin`` role to all federated users.
+
+.. code:: yaml
+
+    mapping:
+     version: RAX-1
+     rules:
+      - local:
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: It looks like there's a mix of 1 and 2 space indents here.

File: doc/attribmapping-basics/faws-mapping.rst [hunk 2]
```diff
@@ -1,9 +1,149 @@
...
+          - path: |
+              (
+                if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='mycompany.123456789012.admin') then ('arn:aws:iam::aws:policy/AdministratorAccess') else ()
+              )
+            multiValue: true
+
+Complete Mapping Policy Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example combines both Fanatical Support for AWS permissions and Fanatical Support for AWS permissions into a single mapping policy.
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Did you mean one of these to just be "AWS permissions"?

File: doc/attribmapping-basics/faws-mapping.rst [hunk 3]
```diff
@@ -1,9 +1,149 @@
...
+            iamPolicies:*:
+              - "arn:aws:iam::aws:policy/job-function/ViewOnlyAccess"
+            iamPolicies:123456789012:
+              - "arn:aws:iam::aws:policy/AdministratorAccess"
+
+As with Fanatical Support for AWS permissions, it is much more common to assign roles conditionally based on a user's group membership. The following mapping policy assigns the following permissions:
+
+* Users in the ``mycompany.global.admin`` group are assigned the ``Administrator`` IAM policy on all AWS accounts
+* Users in the ``mycompany.global.observer`` group are assigned the ``ViewOnly`` IAM policy on all AWS accounts.
+* Users in the ``mycompany.12345678012.admin`` group are assigned the ``Administrator`` IAM policy to a single AWS account, 123456789012.
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Since you're quoting `Administrator`, should you make it exactly the real policy name `AdministratorAccess`?

File: doc/attribmapping-basics/faws-mapping.rst [hunk 4]
```diff
@@ -1,9 +1,149 @@
...
+            name: "{D}"
+          aws:
+            iamPolicies:*:
+              - "arn:aws:iam::aws:policy/job-function/ViewOnlyAccess"
+            iamPolicies:123456789012:
+              - "arn:aws:iam::aws:policy/AdministratorAccess"
+
+As with Fanatical Support for AWS permissions, it is much more common to assign roles conditionally based on a user's group membership. The following mapping policy assigns the following permissions:
+
+* Users in the ``mycompany.global.admin`` group are assigned the ``Administrator`` IAM policy on all AWS accounts
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Since you're quoting `Administrator`, should you make it exactly the real policy name `AdministratorAccess`?

File: doc/attribmapping-basics/faws-mapping.rst [hunk 5]
```diff
@@ -1,9 +1,149 @@
...
+* Users in the ``mycompany.global.observer`` group are assigned the ``ViewOnly`` IAM policy on all AWS accounts.
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Maybe this one should explicitly say `arn:aws:iam::aws:policy/job-function/ViewOnlyAccess` too?

File: doc/attribmapping-basics/faws-mapping.rst [hunk 6]
```diff
@@ -1,9 +1,149 @@
...
 
-Federation for Fanatical Support for AWS is currently unavailable but is under
-development. Instructions for setting up Federation will be available after it
-is released to all customers.
+**Note:** Identity Federation for Fanatical Support for AWS is available as part of a **public beta**. It is available to all customers, but phone support is not available for federated users. All support requests will need to be submitted via the `Rackspace ticketing portal <https://portal.rackspace.com/tickets>`_.
 
+Fanatical Support for AWS Permissions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These permissions control access to features within Rackspace's Fanatical Support for AWS Control Panel. User permissions can be set to ``observer``, ``admin``, or omitted from the mapping policy. Users with ``observer`` permissions have read-only access to the control panel. Users with ``admin`` permissions have read and write access to the control panel. These permissions are assigned by granting them as roles in a mapping policy. The following mapping policy assigns the ``admin`` role to all federated users.
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Should we link back to https://staging.developer.rackspace.com/build-1fc7e2b12f/docs/rackspace-federation/attribmapping-basics/full-roles/ here, just so folks understand what these roles do, and how wide they are?

File: doc/attribmapping-basics/faws-mapping.rst [hunk 7]
```diff
@@ -1,9 +1,149 @@
...
+                if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='mycompany.global.admin') then ('admin') else (),
+                if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='mycompany.global.observer') then ('observer') else ()
+              )
+            multiValue: true
+
+
+AWS Console and API Permissions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These permissions control access to the Amazon Web Services APIs and to features within the AWS Web Console. The following mapping policy assigns all users the "ViewOnly" IAM policy for all AWS accounts. It also assigns the "Administrator" IAM policy to a single AWS account.
```
Comment (adampickeral) [inline doc/attribmapping-basics/faws-mapping.rst]: should this last sentence read
```
It also assigns the ``Administrator`` IAM policy to all users for a single AWS account.
```

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/build-10d3f0fff2/docs/rackspace-federation/)

File: doc/attribmapping-basics/faws-mapping.rst (hunk 1 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed (I think).

File: doc/attribmapping-basics/faws-mapping.rst (hunk 4 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Ya, I realized I used the wrong policy name all over the place. Doh!

File: doc/attribmapping-basics/faws-mapping.rst (hunk 1 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: I fixed it in the source, but it doesn't look like it works in the rendered output. I'll need to dig into that later.

Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Edit: for some reason, Nexus builds aren't reflecting my most recent changes. I suspect once I figure that out, the indentation will be rendered correctly.

**Comment by rackernexus** [comment]
:eyes: [Preview](https://staging.developer.rackspace.com/build-b670b608e2/docs/rackspace-federation/)

File: doc/attribmapping-basics/faws-mapping.rst (hunk 2 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 3 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 4 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 5 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: The AWS console shows the name as "ViewOnlyAccess", so I'd like to match that if you don't feel strongly.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 7 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 1 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst [hunk 8]
```diff
@@ -1,9 +1,192 @@
...
-Fanatical Support for AWS
------------------------------------------------
+===============================================
+Assigning Fanatical Support for AWS Permissions
+===============================================
 
-Federation for Fanatical Support for AWS is currently unavailable but is under
-development. Instructions for setting up Federation will be available after it
-is released to all customers.
+**Note:** Identity Federation for Fanatical Support for AWS is available as part of a **public beta**. It is available to all customers, but phone support is not available for federated users. All support requests will need to be submitted via the `Rackspace ticketing portal <https://portal.rackspace.com/tickets>`_.
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Is it worth tacking on "by a non-federated user." to this sentence, just to be extra, extra clear? I've been reading DD docs this morning and suffering from lack of clarity, so I may be biased.... 1/10 🦆

File: doc/attribmapping-basics/faws-mapping.rst [hunk 9]
```diff
@@ -1,9 +1,192 @@
...
+            email: "{Pt(/saml2p:Response/saml2:Assertion/saml2:Subject/saml2:NameID)}"
+            expire: "PT12H"
+            name: "{D}"
+          aws:
+            iamPolicies:*:
+              - "arn:aws:iam::aws:policy/job-function/ViewOnlyAccess"
+            iamPolicies:123456789012:
+              - "arn:aws:iam::aws:policy/AdministratorAccess"
+
+As with Fanatical Support for AWS permissions, it is much more common to assign roles conditionally based on a user's group membership. The following mapping policy assigns the following permissions:
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Picking nits, but the reduplication of `following` here has me tempted to reword this to avoid repeating that word... "The mapping policy below assigns permissions as follows:" or maybe "Below is an example of mapping policy, and an explanation of the corresponding permissions assigned." 2/10 🦆

File: doc/attribmapping-basics/faws-mapping.rst [hunk 10]
```diff
@@ -1,9 +1,192 @@
...
+
+* Users in the ``mycompany.global.admin`` group are assigned the ``AdministratorAccess`` IAM policy on all AWS accounts
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Missing a sentence-final `.` to match the other items in the list.

File: doc/attribmapping-basics/faws-mapping.rst [hunk 11]
```diff
@@ -1,9 +1,192 @@
...
+          - path: |
+              (
+                if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='mycompany.123456789012.admin') then ('arn:aws:iam::aws:policy/AdministratorAccess') else ()
+              )
+            multiValue: true
+
+AWS Account Creator Permissions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This permissions controls whether or not a user can create new AWS accounts through the Fanatical Support for AWS Control Panel. The following mapping policy grants users in the ``mycompany.global.admin`` group permission to create new AWS accounts:
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Maybe a typo here? `s/This/These/` or `s/permissions/permission/`?

File: doc/attribmapping-basics/faws-mapping.rst [hunk 12]
```diff
@@ -1,9 +1,192 @@
...
+      - local:
+          user:
+            domain: "{D}"
+            email: "{Pt(/saml2p:Response/saml2:Assertion/saml2:Subject/saml2:NameID)}"
+            expire: "PT12H"
+            name: "{D}"
+            roles:
+              - "admin"
+
+It will be much more common to assign roles conditionally based on a user's group membership. The following mapping policy assigns roles conditionally based upon a user's group membership. Users who belong to the ``mycompany.global.admin`` group are granted the ``admin`` role . Users who belong to the ``mycompany.global.observer`` group are granted the ``observer`` role.
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Picking nits again: Is it more common, or will it be more common? Maybe sticking with present tense (over future perfect "will be") makes this a little easier to read?

Comment (martinb3) [reply doc/attribmapping-basics/faws-mapping.rst]: Some of the other examples also just say "it is much more common"

File: doc/attribmapping-basics/faws-mapping.rst (hunk 8 above)
Comment (adampickeral) [reply doc/attribmapping-basics/faws-mapping.rst]: can federated users not submit tickets?

File: doc/attribmapping-basics/faws-mapping.rst [hunk 13]
```diff
@@ -1,9 +1,192 @@
...
+            multiValue: false
+      # Map groups to IAM policies for all AWS accounts
+      - local:
+          aws:
+            iamPolicies:*:
+              - "{0}"
+        remote:
+          - path: |
+              (
+                if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='mycompany.global.admin') then ('arn:aws:iam::aws:policy/AdministratorAccess')
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: is this missing a comma at the end of the line, as well as an `else ()`? When I was writing some of these mappings, the `else ()` didn't seem to be optional.

File: doc/attribmapping-basics/faws-mapping.rst [hunk 14]
```diff
@@ -1,9 +1,192 @@
...
+            name: "{D}"
+          aws:
+            iamPolicies:*:
+              - "{0}"
+            iamPolicies:123456789012:
+              - "{1}"
+        remote:
+          - path: |
+              (
+                if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='mycompany.global.admin') then ('arn:aws:iam::aws:policy/AdministratorAccess')
```
Comment (martinb3) [inline doc/attribmapping-basics/faws-mapping.rst]: Same as below, is this missing an `else ()` and a conditional-final comma?

File: doc/attribmapping-basics/faws-mapping.rst (hunk 8 above)
Comment (martinb3) [reply doc/attribmapping-basics/faws-mapping.rst]: I guess I was assuming we wouldn't be able to call them back or actually help much. Maybe this isn't true.

Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: No, they can definitely get support. It just needs to happen via ticket.

Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Updated.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 12 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Updated.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 9 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 10 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 14 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 11 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 13 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 6 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: My only concern is that the page tends to be targeted at Rackspace Cloud users and includes all sorts of information about Cloud-specific products. What about a link back to our Product Guide instead?

Comment (martinb3) [reply doc/attribmapping-basics/faws-mapping.rst]: That works too 👍

Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Here's the relevant product guide page -- https://manage.rackspace.com/aws/docs/product-guide/access_and_permissions/user_management_and_permissions.html.

**Comment by rackernexus** [comment]
If you want to double-check your formatting, check [the preview](https://staging.developer.rackspace.com/build-28de3eaa96/docs/rackspace-federation/).

**Comment by rackernexus** [comment]
If you want to double-check your formatting, check [the preview](https://staging.developer.rackspace.com/build-ffc921318a/docs/rackspace-federation/).

File: doc/attribmapping-basics/faws-mapping.rst [hunk 15]
```diff
@@ -1,9 +1,196 @@
...
+            email: "{Pt(/saml2p:Response/saml2:Assertion/saml2:Subject/saml2:NameID)}"
+            expire: "PT12H"
+            name: "{D}"
+          aws:
+            iamPolicies:*:
+              - "arn:aws:iam::aws:policy/job-function/ViewOnlyAccess"
+            iamPolicies:123456789012:
+              - "arn:aws:iam::aws:policy/AdministratorAccess"
+
+As with Fanatical Support for AWS permissions, it is much more common to assign roles conditionally based on a user's group membership. The mapping policy below assigns permissions as follows:
```
Comment (smayberry) [inline doc/attribmapping-basics/faws-mapping.rst]: `assign roles` should be `assign IAM policies` or similar?

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/build-20f6a43be5/docs/rackspace-federation/)

File: doc/attribmapping-basics/faws-mapping.rst (hunk 15 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst [hunk 16]
```diff
@@ -1,9 +1,196 @@
...
-Fanatical Support for AWS
------------------------------------------------
+===============================================
+Assigning Fanatical Support for AWS Permissions
+===============================================
 
-Federation for Fanatical Support for AWS is currently unavailable but is under
-development. Instructions for setting up Federation will be available after it
-is released to all customers.
+**Note:** Identity Federation for Fanatical Support for AWS is available as part of a **public beta**. It is available to all customers, but phone support is not available for federated users. Federated users will need to be submit all support requests via the `Rackspace ticketing portal <https://portal.rackspace.com/tickets>`_.
```
Comment (gabecase) [inline doc/attribmapping-basics/faws-mapping.rst]: this is a useful note, @rtgoodwin is there any plans in the near future for the phone support experience to change for federated users?

File: doc/attribmapping-basics/faws-mapping.rst [hunk 17]
```diff
@@ -1,9 +1,196 @@
...
+
+These permissions control access to features within Rackspace's Fanatical Support for AWS Control Panel. User permissions can be set to ``observer``, ``admin``, or omitted from the mapping policy. Users with ``observer`` permissions have read-only access to the control panel. Users with ``admin`` permissions have read and write access to the control panel. These permissions are assigned by granting them as roles in a mapping policy. The following mapping policy assigns the ``admin`` role to all federated users.
+
+.. code:: yaml
+
+  mapping:
+    version: RAX-1
+    rules:
+      - local:
+          user:
```
[the remark this answers is not in the document]
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Once we've migrated customers who are using it currently to the new policy, we'll remove docs and code related to that key. We won't honor that key for any new IdF customers.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 16 above)
Comment (rtgoodwin) [reply doc/attribmapping-basics/faws-mapping.rst]: @gabecase In roadmap, pending approval. For now any mention should be "not currently supported".

Comment (gabecase) [reply doc/attribmapping-basics/faws-mapping.rst]: @bradgignac some nits to shorten: 
`It is available` can be shortened to `It's available`
`phone support is not available` can be shortened to `phone support isn't available`

File: doc/attribmapping-basics/faws-mapping.rst [hunk 18]
```diff
@@ -1,9 +1,196 @@
...
+            domain: "{D}"
+            email: "{Pt(/saml2p:Response/saml2:Assertion/saml2:Subject/saml2:NameID)}"
+            expire: "PT12H"
+            name: "{D}"
+            roles:
+              - "admin"
+
+It is much more common to assign roles conditionally based on a user's group membership. The following mapping policy assigns roles conditionally based upon a user's group membership. Users who belong to the ``mycompany.global.admin`` group are granted the ``admin`` role . Users who belong to the ``mycompany.global.observer`` group are granted the ``observer`` role.
```
Comment (gabecase) [inline doc/attribmapping-basics/faws-mapping.rst]: `It is much more...` can be shortened to `It's`

File: doc/attribmapping-basics/faws-mapping.rst [hunk 19]
```diff
@@ -1,9 +1,196 @@
...
+            email: "{Pt(/saml2p:Response/saml2:Assertion/saml2:Subject/saml2:NameID)}"
+            expire: "PT12H"
+            name: "{D}"
+          aws:
+            iamPolicies:*:
+              - "arn:aws:iam::aws:policy/job-function/ViewOnlyAccess"
+            iamPolicies:123456789012:
+              - "arn:aws:iam::aws:policy/AdministratorAccess"
+
+As with Fanatical Support for AWS permissions, it is much more common to assign IAM policies conditionally based on a user's group membership. The mapping policy below assigns permissions as follows:
```
Comment (gabecase) [inline doc/attribmapping-basics/faws-mapping.rst]: `it is much more` can be shortened to `it's much more`

**Comment by bradgignac** [comment]
@gabecase I intentionally avoided contractions because of yeas of English/technical writing classes . Do we have a general style recommendation for or against contractions for our customer-facing documentation?

**Comment by gabecase** [comment]
You might consider contributing some example yaml policies directly to https://github.com/rackerlabs/fed-policy-examples/tree/master/ which does some automated linting on the examples we keep in there. Identity is planning on migrating this over to xml in the near future though so i'd be best to keep it simple until then if you share any.

Formatting yaml in rst is kind of a pain so that repository has come in handy when we wanted to point a racker/customer to specific example they could download or specific lines in a policy.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 16 above)
Comment (RackerKate) [inline doc/attribmapping-basics/faws-mapping.rst]: Suggest "...Federated users need to submit all support requests through..."

**Comment by gabecase** [comment]
@bradgignac https://developer.rackspace.com/docs/style-guide/style/contractions/ thx to @RackerKate for digging that up! It's more of a 'you can use them if you think it's appropriate' but if you prefer to keep it than I'm fine.

File: doc/attribmapping-basics/faws-mapping.rst [hunk 20]
```diff
@@ -1,9 +1,196 @@
...
 
-Federation for Fanatical Support for AWS is currently unavailable but is under
-development. Instructions for setting up Federation will be available after it
-is released to all customers.
+**Note:** Identity Federation for Fanatical Support for AWS is available as part of a **public beta**. It is available to all customers, but phone support is not available for federated users. Federated users will need to be submit all support requests via the `Rackspace ticketing portal <https://portal.rackspace.com/tickets>`_.
 
+Fanatical Support for AWS Permissions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These permissions control access to features within Rackspace's Fanatical Support for AWS Control Panel. User permissions can be set to ``observer``, ``admin``, or omitted from the mapping policy. Users with ``observer`` permissions have read-only access to the control panel. Users with ``admin`` permissions have read and write access to the control panel. These permissions are assigned by granting them as roles in a mapping policy. The following mapping policy assigns the ``admin`` role to all federated users.
```
Comment (RackerKate) [inline doc/attribmapping-basics/faws-mapping.rst]: Suggest capitalizing all instances of "control panel" as "Control Panel" for consistency.
Suggest replacing the period at the end of the paragraph with a colon (:).

File: doc/attribmapping-basics/faws-mapping.rst (hunk 18 above)
Comment (RackerKate) [inline doc/attribmapping-basics/faws-mapping.rst]: Suggest "It's much more common to assign roles conditionally based on a user's group membership. In the following mapping policy, users who belong to the ``mycompany.global.admin`` group are granted the ``admin`` role:"

File: doc/attribmapping-basics/faws-mapping.rst (hunk 16 above)
Comment (RackerKate) [reply doc/attribmapping-basics/faws-mapping.rst]: @gabecase Those contractions are correct, but in this particular context I think it reads best to keep the text as-is.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 18 above)
Comment (RackerKate) [reply doc/attribmapping-basics/faws-mapping.rst]: Agreed for this context. This will read well.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 19 above)
Comment (RackerKate) [reply doc/attribmapping-basics/faws-mapping.rst]: Agreed.

File: doc/attribmapping-basics/faws-mapping.rst [hunk 21]
```diff
@@ -1,9 +1,196 @@
...
+            iamPolicies:*:
+              - "arn:aws:iam::aws:policy/job-function/ViewOnlyAccess"
+            iamPolicies:123456789012:
+              - "arn:aws:iam::aws:policy/AdministratorAccess"
+
+As with Fanatical Support for AWS permissions, it is much more common to assign IAM policies conditionally based on a user's group membership. The mapping policy below assigns permissions as follows:
+
+* Users in the ``mycompany.global.admin`` group are assigned the ``AdministratorAccess`` IAM policy on all AWS accounts.
+* Users in the ``mycompany.global.observer`` group are assigned the ``ViewOnlyAccess`` IAM policy on all AWS accounts.
+* Users in the ``mycompany.12345678012.admin`` group are assigned the ``AdministratorAccess`` IAM policy to a single AWS account, 123456789012.
```
Comment (RackerKate) [inline doc/attribmapping-basics/faws-mapping.rst]: Suggest "...only for AWS account 123456789012."

File: doc/attribmapping-basics/faws-mapping.rst [hunk 22]
```diff
@@ -1,9 +1,196 @@
...
+                if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='mycompany.global.observer') then ('arn:aws:iam::aws:policy/job-function/ViewOnlyAccess') else ()
+              )
+            multiValue: true
+          - path: |
+              (
+                if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='mycompany.123456789012.admin') then ('arn:aws:iam::aws:policy/AdministratorAccess') else ()
+              )
+            multiValue: true
+
+Many customers create their own `customer managed policies <https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_managed-vs-inline.html#customer-managed-policies>`_ that are the same across many AWS accounts. In order to easily assign this policy, policy ARNs can omit the account ID section of the ARN. For example, there might be a policy named ``MyCompany.Audit`` that exists on every AWS account. To assign this policy, simply use ``arn:aws:iam:::policy/MyCompany.Audit`` in your mapping policy.
```
Comment (RackerKate) [inline doc/attribmapping-basics/faws-mapping.rst]: Suggest "...Policy ARNs can omit the account ID section, which makes it easier to assign these policies. For example, if a policy named ``MyCompany.Audit`` exists on every AWS account, you can assign this policy by using ``arn:aws:iam:::policy/MyCompany.Audit`` in your mapping policy."

File: doc/attribmapping-basics/faws-mapping.rst [hunk 23]
```diff
@@ -1,9 +1,196 @@
...
+          - path: |
+              (
+                if (mapping:get-attributes('http://schemas.xmlsoap.org/claims/Group')='mycompany.global.admin') then ('true') else ('false')
+              )
+            multiValue: false
+
+Complete Mapping Policy Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example combines both Fanatical Support for AWS permissions and AWS Console and API permissions into a single mapping policy.
```
Comment (RackerKate) [inline doc/attribmapping-basics/faws-mapping.rst]: Suggest replacing the period at the end of the paragraph with a colon (:).

File: doc/attribmapping-basics/faws-mapping.rst (hunk 18 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Your suggestion omits the mention of `mycompany.global.observer`. How about the following:

> It's much more common to assign roles conditionally based on a user's group membership. In the following mapping policy, users who belong to the ``mycompany.global.admin`` group are granted the ``admin`` role, and users who belong to the ``mycompany.global.observer`` group are granted the ``observer`` role:

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/build-016c943d76/docs/rackspace-federation/). :bow:

File: doc/attribmapping-basics/faws-mapping.rst (hunk 16 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 20 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 18 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Updated to my suggestion that includes mention of the observer role.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 21 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 22 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst (hunk 23 above)
Comment (bradgignac) [reply doc/attribmapping-basics/faws-mapping.rst]: Fixed.

File: doc/attribmapping-basics/faws-mapping.rst [hunk 24]
```diff
@@ -1,9 +1,196 @@
...
+              )
+            multiValue: true
+
+For more information about Fanatical Support for AWS permissions, visit the `User Management and Permissions <https://manage.rackspace.com/aws/docs/product-guide/access_and_permissions/user_management_and_permissions.html>`_ section of the Fanatical Support for AWS product guide.
+
+AWS Console and API Permissions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+These permissions control access to the Amazon Web Services APIs and to features within the AWS Web Console. The following mapping policy assigns all users the "ViewOnlyAccess" IAM policy for all AWS accounts. It also assigns the "AdministratorAccess" IAM policy to all users for a single AWS account.
+
```
Comment (ltvolks) [inline doc/attribmapping-basics/faws-mapping.rst]:
```suggestion

.. note:: In order for AWS IAM policy mappings to take effect, the user must have either ``admin`` or ``observer`` Fanatical Support for AWS Permissions roles described in the previous section.

```

File: doc/attribmapping-basics/faws-mapping.rst (hunk 18 above)
Comment (gabecase) [reply doc/attribmapping-basics/faws-mapping.rst]: makes sense to me 👍

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-db2c332b0e/docs/rackspace-federation/)
